src/app/session/session-state.service.ts

```diff
@@ -27,16 +10,23 @@ export class SessionState {
   session = new BehaviorSubject<Session>(new Session());
 
   get budgetItems(): Observable<BudgetItem[]> {
-    const sessionObservable = this.session.asObservable();
+    const session$ = this.session.asObservable();
 
-    return sessionObservable.pipe(map(session => session.budgetItems));
+    return session$.pipe(map(session => session.budgetItems));
   }
+
   get categories(): Observable<Category[]> {
-    const sessionObservable = this.session.asObservable();
+    const session$ = this.session.asObservable();
 
-    return sessionObservable.pipe(map(session => session.categories));
+    return session$.pipe(map(session => session.categories));
   }
 
+  getCategory(name: string): Category {
+    const session = this.session.getValue();
+
+    const category = session.categories.find(item => item.name === name);
+    return { ...category };
```

Comment:
this clones the `category` so that any changes made to the returned value don't effect the service

Reply:
Before, when we were just using the getter with the .pipe method, we were changing the actual value of the category?

Reply:
I believe getValue() returns a copy of the current value, so we should be okay there

Reply:
So then why was this change necessary?

Reply:
This is a new function, that gets a specific category when provided with the category's name. If you don't care about the other categories, and only want the current value this is useful

Reply:
Ahh ok. I see.